lib/algos/list/delimiterMaster.js

```diff
@@ -1,84 +1,110 @@
+ * to return the raw master versions of existing objects.
  */
 class DelimiterMaster extends Delimiter {
     /**
-     * Overriding the base function to extract the versionId of the entry.
-     *
-     * @param {string} key - the key of the entry
-     * @param {object} value - the value of the entry
-     * @return {undefined}
+     * Delimiter listing of master versions.
+     * @param {Object} parameters           - listing parameters
+     * @param {String} parameters.delimiter - delimiter per amazon format
+     * @param {String} parameters.prefix    - prefix per amazon format
+     * @param {String} parameters.marker    - marker per amazon format
+     * @param {Number} parameters.maxKeys   - number of keys to list
+     */
+    constructor(parameters) {
+        super(parameters);
+        this.PHD = undefined;
+    }
+
+    /**
+     *  Add a (key, value) tuple to the listing
+     *  Set the NextMarker to the current key
+     *  Increment the keys counter
+     *  @param {String} key   - The key to add
+     *  @param {String} value - The value of the key
+     *  @return {number}      - indicates if iteration should continue
      */
     addContents(key, value) {
-        this.Contents.push({
-            key,
-            value: {
-                Size: value['content-length'],
-                ETag: value['content-md5'],
-                LastModified: value['last-modified'],
-                // <versioning>
-                VersionId: VSUtils.getts(value),
-                // </versioning>
-                Owner: {
-                    DisplayName: value['owner-display-name'],
-                    ID: value['owner-id'],
-                },
-                StorageClass: value['x-amz-storage-class'],
-                Initiated: value.initiated,
-                Initiator: value.initiator,
-                EventualStorageBucket: value.eventualStorageBucket,
-                partLocations: value.partLocations,
-                creationDate: value.creationDate,
-            },
-        });
+        if (this._reachedMaxKeys()) {
+            return -1;
+        }
+        this.Contents.push({ key, value });
         this.NextMarker = key;
         ++this.keys;
+        return 1;
     }
 
     /**
-     * Overriding the filter function that formats the
-     * listing results based on the listing algorithm.
-     *
-     * @param {object} obj - metadata entry in the form of { key, value }
-     * @return {boolean} - continue filtering or return the formatted list
+     *  Filter to apply on each iteration, based on:
+     *  - prefix
+     *  - delimiter
+     *  - maxKeys
+     *  The marker is being handled directly by levelDB
+     *  @param {Object} obj       - The key and value of the element
+     *  @param {String} obj.key   - The key of the element
+     *  @param {String} obj.value - The value of the element
+     *  @return {Boolean}         - indicates if iteration should continue
```

Comment:
well, it's fine if that's the reason, but the generable JSDOC won't show it outright. minor anyways